common/lib/xmodule/xmodule/x_module.py

```diff
@@ -398,7 +398,7 @@ def get_content_titles(self):
         the display_names of all its leaf nodes.
         """
         if self.has_children:
-            return sum((child.get_content_titles() for child in self.get_children()), [])
+            return sum((child.get_content_titles() for child in self.get_display_items()), [])
```

Comment:
As per my understanding and debugging, ```has_access``` is being applied to ```get_display_items``` method. This [line](https://github.com/edx/edx-platform/blob/99ac9912ec517f1f958b79dd32b436d609debcea/common/lib/xmodule/xmodule/x_module.py#L1059) sets indirection on ```get_display_items``` due to which ```has_access``` will be called and it solves the issue.

Below is the call stack 

```
can_load access.py:396                                      
_dispatch access.py:512                                     
_has_access_descriptor access.py:431                        
has_access access.py:89                                     
get_module_for_descriptor_internal module_render.py:596     
inner_get_module module_render.py:341                       
get_children [SequenceModuleWithMixins] x_module.py:671     
get_display_items [SequenceModuleWithMixins] x_module.py:447
```